### PR TITLE
Add functions for getting indices of features of different types to C API

### PR DESCRIPTION
**Pull request by nimusp**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en.

I am adding methods for retrieving feature indices.
The Python API already has it, but the C API is missing it.

### CODE REVIEW

File: catboost/libs/model/model.cpp [hunk 1]
```diff
@@ -1341,6 +1341,46 @@ TVector<TString> GetModelUsedFeaturesNames(const TFullModel& model) {
...
+
+TVector<int> GetModelEmbeddingFeaturesIndices(const TFullModel& model) {
+    const auto features = model.ModelTrees->GetEmbeddingFeatures();
+    TVector<int> indices;
+    indices.reserve(features.size());
+    for (const auto& feature : features) {
+        indices.push_back(feature.Position.FlatIndex);
+    }
+    return indices;
+}
```
Comment (andrey-khropov) [inline catboost/libs/model/model.cpp:1382]: Please avoid copy-paste and create a helper function that will create TVector<int> from the result of Get*Features functions.

File: catboost/libs/model_interface/c_api.cpp [hunk 2]
```diff
@@ -888,5 +888,61 @@ CATBOOST_API bool GetModelUsedFeaturesNames(ModelCalcerHandle* modelHandle, char
...
+    if (!*indices) {
+        return false;
+    }
+
+    *count = featureIndices.size();
+    for (size_t i = 0; i != featureIndices.size(); ++i) {
+        (*indices)[i] = featureIndices[i];
+    }
+    return true;
+}
```
Comment (andrey-khropov) [inline catboost/libs/model_interface/c_api.cpp:945]: Please avoid copy-paste and create a helper function that will copy the data from TVector to `indices` and `count`.

Comment (andrey-khropov) [inline catboost/libs/model_interface/c_api.cpp:945]: Please wrap GetModel*FeaturesIndices call in try { } catch block and set an error message and return false if the exception is cached (see similar logic in other functions in this file).

**Review by andrey-khropov — CHANGES_REQUESTED** [review]

File: catboost/libs/model/model.cpp (hunk 1 above)
Comment (nimusp) [reply catboost/libs/model/model.cpp]: There may be a problem with converting to the parent (TBaseArrayFeature) when using TConstArrayRef, so I added a workaround with templates

**Review by andrey-khropov — APPROVED** [review]

**Pull request closed without merging**